File: src/main/java/Vacation.java

public class Vacation
{
	public int start;
	public int stop;
	public int breakStart;
	public int breakStop;
	public int cost;

	public Vacation()
	{
		start = stop = breakStart = breakStop = stop = 0;
	}
	
	public Vacation(final Vacation anotherVacation)
	{
		start = anotherVacation.start;
		stop = anotherVacation.stop;
		breakStart = anotherVacation.breakStart;
		breakStop = anotherVacation.breakStop;
		cost = anotherVacation.cost;
	}

	public boolean isWorkedAt (int t){
		return ((start <= t) && (t < breakStart)) || ((breakStart <= t) && (t < breakStop));
	}
	
	public boolean check()
	{
		return
			   (start <= breakStart)
			&& (breakStart <= breakStop)
			&& (breakStop <= stop);
	}
	
	public String visualizeVacation(final int nbPeriods)
	{
		String result = "";
		for(int period = 1; period < start; period++)
		{
			result += "    ";
		}
		for(int period = start; period <= stop; period++)
		{
			if(period >= breakStart && period <= breakStop)
			{
				result += "|~~~";
			}
			else
			{
				result += "|---";
			}
		}
		result += "|   ";
		for(int period = stop+2; period <= nbPeriods; period++)
		{
			result += "    ";
		}
		return result;
	}
}

File: cplusplus/vacation.cpp
#include "vacation.h"

using namespace std;

unsigned int Vacation::numberOfPeriods = 0;

Vacation::Vacation(const unsigned int start, const unsigned int stop, const unsigned int breakStart, const unsigned int breakStop) :
    start(start),
    stop(stop),
    breakStart(breakStart),
    breakStop(breakStop)
{
    if(!check())
    {
        throw invalid_argument("Invalid periods [Vacation constructor]");
    }
}

bool Vacation::isWorkedAt(const unsigned int period) const
{
    return ((start <= period) && (period < breakStart)) || ((breakStop < period) && (period <= stop));
}

bool Vacation::check() const
{
    return
           (start <= breakStart)
        && (breakStart <= breakStop)
        && (breakStop <= stop);
}

ostream & operator << (ostream & output, const Vacation & vacation)
{
    unsigned int period = 0;
    for(/* period = 0 */; period < vacation.start; ++period)
        output << "    ";
    for(/* period = vacation.start */; period < vacation.breakStart; ++period)
        output << "|---";
    for(/* period = vacation.breakStart */; period <= vacation.breakStop; ++period)
        output << "||||";
    for(/* period = vacation.breakStop + 1 */; period <= vacation.stop; ++period)
        output << "|---";
    // period == vacation.stop + 1
        output << "|   ";
    for(++period /* period = vacation.stop + 2 */; period <= Vacation::numberOfPeriods; ++period)
        output << "    ";
    return output;
}
